### PR TITLE
Nightly Hue firmware update

### LINKED ISSUE #38: Bridge firmware update command
Bridge firmware can be updated with some REST calls, would be nice to implement support for that.

### PR DESCRIPTION
**Pull request by praseodym**
Closes #38

### CODE REVIEW

File: src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java [hunk 1]
```diff
@@ -236,4 +242,54 @@ public void updateLightState(HueLamp lamp, HueLightState lightState) throws Brid
...
+            log.info("Bridge update completed");
+            rt.put(bridgeConfigUrl, "{\"swupdate\": {\"notify\":false}}");
+        } else if (swUpdate.getUpdatestate() == 0) {
+            log.info("Initiating bridge update check");
+            rt.put(bridgeConfigUrl, "{\"swupdate\": {\"checkforupdate\":true}}");
+        }
+        if (swUpdate.getUpdatestate() == 2) {
+            log.info("Installing bridge update: {}", swUpdate.getText());
+            rt.put(bridgeConfigUrl, "{\"swupdate\": {\"updatestate\":3}}");
+        }
```
Comment (RickWieman) [inline src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java:266]: Isn't it cleaner to use `JSONObject`s here? Just wondering, as we also do that for strobe (https://github.com/WISVCH/chue/blob/master/src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java#L168-L173)...

**Comment by RickWieman** [comment]
Other than that, :+1:

File: src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java (hunk 1 above)
Comment (praseodym) [inline src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java:266]: The `PHHueHttpConnection` class isn't documented so that's why I didn't want to use it.

Comment (RickWieman) [inline src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java:266]: You don't need that in order to rewrite those `String`s to `JSONObject`s right? If you think using `String`s is cleaner, then we'll just use this...

Comment (praseodym) [inline src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java:266]: `JSONObject` is also one of the weird Hue classes :disappointed:

**Comment by RickWieman** [comment]
:shipit:

**Pull request merged**